src/lib/shim/src/lib.rs

```diff
@@ -544,7 +544,15 @@ fn wait_for_start_event(get_initial_working_dir: bool) {
         ipc.to_shadow().send(start_req);
         ipc.from_shadow().receive().unwrap()
     });
-    assert!(matches!(res, ShimEventToShim::StartRes));
+    let ShimEventToShim::StartRes(res) = res else {
+        panic!("Unexpected response: {res:?}");
+    };
+    if is_first_thread {
+        // SAFETY: We're ensuring serial execution in this process, and no other
+        // Rust code in this library should have tried accessing the auxiliary
+        // vector yet, so no references should exist.
```

Comment:
Maybe note that we assume/hope plugin code hasn't run yet and accessed or made a constant reference to the `AT_RANDOM` data? And that the loader may have already used this value, but that's fine?